src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -237,6 +250,39 @@ RestExecuteOnExtensionResponse handleRestExecuteOnExtensionRequest(RestExecuteOn
                 "No handler for " + ExtensionRestPathRegistry.restPathToString(request.getMethod(), request.getUri())
             );
         }
+        // Action name here will be prefix (extensions:{extensionName} + route + method.
+        // For an extensions named `hw`, a call to /hello yields an action name of extensions:hw/hello/get
+        String actionName = restHandler.getActionPrefix() + request.getUri() + "/" + request.getMethod().toString().toLowerCase();
```

Comment:
we should be able to get the uniqueID for the extension to use as the prefix here.   Also suggest `String.join()` with the slashes.  Also I think `toLowerCase()` might run afoul of some OpenSearch conventions banning its use (for localization or something).  Shouldn't be an issue for UTF-8 compatible Method enum, though...

Reply:
This is really the crux of what I want to propose for this prototype of authorization. As far as I can see in core, each action is handled by its own file that extends `ActionType`. Currently in the SDK we `implement ExtensionRestHandler` and are responsible for figuring out how to handle each individual route that's registered inside `handleRequest`. The `handleRequest` function could end up being huge for an extension with many routes.

By concatenating the prefix (`extensions:{extensionName}`) + route + method this is creating a unique action name which can be used in an authorization request to let core determine ultimately whether the action request is granted or denied. This is the convention I have chosen for this iteration of the Draft PR.

The idea here is a simple AuthorizationRequest using the principalIdentifier, actionName, a list of resourceIds and a Map of additionalParams to keep it simple to start with and the response would have a status of either granted or denied. The logic of determining whether a request is granted or denied will be in a companion PR on the `feature/extensions` branch of core.

We may want to consider other ways extensions can determine what actions a user can perform on a resource as well.

I'm currently working backwards from the AD plugin to determine what is needed in the SDK to authorize actions for the extension. Admittedly, I'm having some writers block in designing a good generic way that extensions developers can use to protect actions and resources so any feedback to the design of the AuthorizationRequest is welcome.

Reply:
> The handleRequest function could end up being huge for an extension with many routes.

There can be multiple implementations of `ExtensionRestHandler`.  There's only one in this sample, but look here for example: https://github.com/opensearch-project/anomaly-detection/tree/main/src/main/java/org/opensearch/ad/rest

Each Action class is the same as our ExtensionRestHandler class.  Some have a few routes, some just have one.  I could have easily left the GET /hello in one action and created a new action for PUT.

> By concatenating the prefix (extensions:{extensionName}) + route + method this is creating a unique action name 

That (with unique ID) the same concatenation I'm using in the RestController.  But extension Name isn't guaranteed to be unique which is why I was suggesting uniqueId.

Reply:
Ok, I see my error now. I conflated `uniqueId` and `extensionName` because I didn't differentiate between the `extensions.yml` defined in the `extensions/extensions.yml` of core and the extensions specific settings (the sample configs in `main/resources/sample/) containing a name, hostAddress and port. 

There may be a problem with defining `uniqueId` in `extensions.yml` instead of the extension settings files. I see why its in `extensions.yml` so admins can namespace the extensions as they see fit (and its used for the base of the API), but plugin permissions usually use a shorthand for the plugin. i.e. `cluster:admin/opendistro/ad/detector/write` where `ad` is the shorthand for anomaly detector. We can keep `uniqueId` so admins can assign them as needed, but we also need static unique identifier for each extension.

Reply:
> We can keep uniqueId so admins can assign them as needed, but we may also need a static unique identifier for each extension.

The fully qualified class name should be unique, assuming reverse DNS package naming standard.